checker/src/main/java/org/checkerframework/checker/index/inequality/LessThanVisitor.java

```diff
@@ -131,4 +138,47 @@ protected boolean isTypeCastSafe(AnnotatedTypeMirror castType, AnnotatedTypeMirr
 
         return super.isTypeCastSafe(castType, exprType);
     }
+
+    // TODO: This belongs in an IndexVisitor.
```

Comment:
Is there some reason not to do this now?

Reply:
There is no `IndexVisitor` class.  We would need to create one just to contain this check.
I wanted to get your opinion about whether that is worth doing.

Reply:
> There is no IndexVisitor class. We would need to create one just to contain this check. That would add a bit of complexity, but more seriously it would add another pass, slowing down the checker.

`IndexVisitor` should be a subclass of `UpperBoundVisitor`. (just like `IndexChecker` is a subclass of `UpperBoundChecker`.). So this wouldn't be a full additional pass.

>I wanted to get your opinion about whether that is worth doing.

So in this pull request, you've only added warnings about the LessThanChecker.  I'm assuming you want to add warnings for all of the other Index Checker annotations.  I can see the benefits of doing that all in one place, but it would also make sense for each checker to issue warnings about its own annotations.

Also, it probably makes sense to generalize this more.  One thought is to add warnings if an annotation is written on a type that is not in the list of its `@RelevantJavaTypes` meta-annotation.  We can discuss more at our next meeting.